src/clj/witan/send/charts.clj

```diff
@@ -107,4 +107,74 @@
                                                                    :values (r-combine colours)}]]
                                              [[:scale_x_continuous {:breaks (r-combine (filter even? (range min-year max-year)))
                                                                     :limit [:c min-year max-year]}]]))]))
-  (move-file "Rplots.pdf" (str "target/" title "_probability.pdf")))
+  (move-file "Rplots.pdf" (str "target/" title "_Probability.pdf")))
+
+(defn get-total-historical-send-pop [year transitions-data]
+  (count (map :academic-year-1 (filter #(= (:calendar-year %) year) (filter #(not= (:setting-1 %) :NONSEND) transitions-data)))))
+
+(defn is-send-and-is-year? [row year]
+  (and (not= (:setting-1 row) :NONSEND)
+        (= (:calendar-year row) year)))
+
+(defn foo [year transitions-data]
+  (->> transitions-data
+       (filter #(is-send-and-is-year? % year))
+       (map :academic-year-1)
+       count))
+
+(defn population-line-plot
+  [historic-data future-projections-data]
+  (let [historic-years (vec (distinct (map :calendar-year historic-data)))
+        projected-years (vec (#(range % (+ (count future-projections-data) %))
+                         (inc (last historic-years))))
+        historic-pops (map #(get-total-historical-send-pop % historic-data) historic-years)
+        years (into [] (concat historic-years projected-years))
+        means (into [] (concat historic-pops (map int (map :mean future-projections-data))))
+        low-cis (into [] (concat (repeat (count historic-years) :NA) (map :low-ci future-projections-data)))
+        high-cis (into [] (concat (repeat (count historic-years) :NA) (map :high-ci future-projections-data)))
+        gg4clj-data-frame {:year years :mean means :min low-cis :max high-cis}
+        min-year (first historic-years)
+        max-year (last projected-years)
+        max-max (apply max (map :max future-projections-data))]
```

Comment:
Is this max population?

Reply:
it sets the upper limit on the y-axis by taking the largest max value. I'll make this more explicit.